Zend/zend_builtin_functions.c

```diff
@@ -1061,6 +1061,9 @@ static inline void _class_exists_impl(zval *return_value, zend_string *name, boo
 	if (ZSTR_HAS_CE_CACHE(name)) {
 		ce = ZSTR_GET_CE_CACHE(name);
 		if (ce) {
+			if (flags == ZEND_ACC_LINKED && ce->ce_flags & ZEND_ACC_ENUM) {
+				zend_error(E_DEPRECATED, "using class_exists() for enums is deprecated, triggerred for calling class_exists() for enum \"%s\". Use enum_exists() instead", ZSTR_VAL(ce->name));
```

Comment:
```suggestion
				zend_error(E_DEPRECATED, "using class_exists() for enums is deprecated, triggered for calling class_exists() for enum \"%s\". Use enum_exists() instead", ZSTR_VAL(ce->name));
```

(here and elsewhere)